pkg/services/thumbs/crawler.go

```diff
@@ -247,3 +237,56 @@ func (r *simpleCrawler) walk() {
 	r.status.Finished = time.Now()
 	r.broadcastStatus()
 }
+
+// TODO extract the three methods below from this file
+func (r *simpleCrawler) SaveThumbnailFromFile(tempFilePath string, dashboardID int64, dashboardUID string, theme rendering.Theme, kind models.ThumbnailKind) (int64, error) {
+	defer func() {
+		err := os.Remove(tempFilePath)
+		if err != nil {
+			tlog.Error("failed to remove thumbnail temp file", "dashboardUID", dashboardUID, "err", err)
+		}
+	}()
+
+	file, err := os.Open(tempFilePath)
+	if err != nil {
+		tlog.Error("error opening file", "dashboardUID", dashboardUID, "err", err)
+		return 0, err
+	}
+
+	reader := bufio.NewReader(file)
+	content, err := ioutil.ReadAll(reader)
+
+	if err != nil {
+		tlog.Error("error reading file", "dashboardUID", dashboardUID, "err", err)
+		return 0, err
+	}
+
+	return r.SaveThumbnailFromBytes(content, r.GetMimeType(tempFilePath), dashboardID, dashboardUID, theme, kind)
+}
+
+func (r *simpleCrawler) GetMimeType(filePath string) string {
```

Comment:
```suggestion
func getMimeType(filePath string) string {
```
no need for this to be a member function (and you can remove it from the interface)